Src/Couchbase.Linq/BucketContext.cs

```diff
@@ -71,7 +73,7 @@ public IQueryable<T> Query<T>()
         /// </value>
         public string BucketName
         {
-            get { return _bucket.Name; }
+            get { return Bucket.Name; }
         }
```

Comment:
Instead of directly exposing the IBucket implementation here, why not instead add a method on IBucketContext  with a signature like this:

    public void/int Execute(string statement);

This will abstract out the underlying SDK semantics making it easier to "maintain" later on. 

Also, note that the two objects have very different lifespans: the IBucketContext impl is an ephemeral object and the CouchbaseBucket is a long-lived, application scoped object...this may cause confusion later on.